src/platform/plugins/shared/data/server/search/session/get_search_status.ts

```diff
@@ -8,28 +8,36 @@
  */
 
 import { i18n } from '@kbn/i18n';
+import type { EsqlAsyncEsqlResult } from '@elastic/elasticsearch/lib/api/types';
+
 import type { TransportResult } from '@elastic/elasticsearch';
 import { ElasticsearchClient } from '@kbn/core/server';
-import { SearchSessionRequestStatus } from '../../../common';
+import { SearchSessionRequestInfo, SearchSessionRequestStatus } from '../../../common';
 import { SearchStatus } from './types';
 import { AsyncSearchStatusResponse } from '../..';
 
 export async function getSearchStatus(
   internalClient: ElasticsearchClient,
-  asyncId: string
+  asyncId: string,
+  session: SearchSessionRequestInfo,
+  logger: Logger,
+  asUserClient: ElasticsearchClient
 ): Promise<SearchSessionRequestStatus> {
-  // TODO: Handle strategies other than the default one
-  // https://github.com/elastic/kibana/issues/127880
+  logger.debug(`SearchSession getSearchStatus ${asyncId}`);
   try {
-    const apiResponse: TransportResult<AsyncSearchStatusResponse> =
-      await internalClient.asyncSearch.status(
-        {
-          id: asyncId,
-        },
-        { meta: true }
-      );
+    const apiResponse: EsqlAsyncEsqlResult | TransportResult<AsyncSearchStatusResponse, unknown> =
+      session.strategy === 'esql_async'
+        ? await asUserClient.esql.asyncQueryGet({ id: asyncId }, { meta: true })
+        : await internalClient.asyncSearch.status(
```

Comment:
all here is necessary for ES|QL support , and it was my last blocker, I got a 404 initially using the internalClient, until @ppisljar told me that all esql queries using the `asCurrentUser` client. Then it worked

FYI @AlexGPlay @lukasolson 

Note that all loggers here wouldn't be necessary, but I used them for debugging after starting with console.logs ... now they are not longer necessary, but on the other hand, in the debugging case? Why not